include/dlflcore/DLFLCommon.hh

```diff
@@ -238,34 +256,52 @@ struct eqstr {
   bool operator()(int a, int b) const { return a == b; }
 };
 
-typedef __gnu_cxx::hash<unsigned int> Hash;
-typedef __gnu_cxx::hash_map<unsigned int, unsigned int, Hash, eqstr> HashMap;
+#ifdef _MSC_VER
+  typedef std::hash<unsigned int> Hash;
+  typedef std::unordered_map<unsigned int, unsigned int, Hash, eqstr> HashMap;
+#else
+  typedef __gnu_cxx::hash<unsigned int> Hash;
+  typedef __gnu_cxx::hash_map<unsigned int, unsigned int, Hash, eqstr> HashMap;
+#endif //_MSC_VER
 
 } // namespace DLFL
 
+
+#ifndef _MSC_VER
 namespace __gnu_cxx {
-template <> struct hash<DLFL::DLFLVertexPtr> {
-  size_t operator()(DLFL::DLFLVertexPtr p) const {
-    return reinterpret_cast<size_t>(p);
-  }
-};
-template <> struct hash<DLFL::DLFLEdgePtr> {
-  size_t operator()(DLFL::DLFLEdgePtr p) const {
-    return reinterpret_cast<size_t>(p);
-  }
-};
-template <> struct hash<DLFL::DLFLFacePtr> {
-  size_t operator()(DLFL::DLFLFacePtr p) const {
-    return reinterpret_cast<size_t>(p);
-  }
-};
+  template <> struct hash<DLFL::DLFLVertexPtr> {
+    size_t operator()(DLFL::DLFLVertexPtr p) const {
+      return reinterpret_cast<size_t>(p);
+    }
+  };
+  template <> struct hash<DLFL::DLFLEdgePtr> {
+    size_t operator()(DLFL::DLFLEdgePtr p) const {
+      return reinterpret_cast<size_t>(p);
+    }
+  };
+  template <> struct hash<DLFL::DLFLFacePtr> {
+    size_t operator()(DLFL::DLFLFacePtr p) const {
+      return reinterpret_cast<size_t>(p);
+    }
+  };
 } // namespace __gnu_cxx
+#endif //_MSC_VER
+
 namespace DLFL {
-typedef __gnu_cxx::hash_set<DLFLVertexPtr> DLFLVertexPtrSet;
-typedef __gnu_cxx::hash_set<DLFLEdgePtr> DLFLEdgePtrSet;
-typedef __gnu_cxx::hash_set<DLFLFacePtr> DLFLFacePtrSet;
-typedef __gnu_cxx::hash_map<DLFLVertexPtr, DLFLVertexPtr> DLFLVertexPtrMap;
-typedef __gnu_cxx::hash_map<DLFLEdgePtr, DLFLEdgePtr> DLFLEdgePtrMap;
-typedef __gnu_cxx::hash_map<DLFLFacePtr, DLFLFacePtr> DLFLFacePtrMap;
+#ifdef _MSC_VER
+    typedef std::unordered_set<DLFLVertexPtr> DLFLVertexPtrSet;
```

Comment:
hash_set and hash_map are depreciated and not supported by the visual studio compiler anymore. I dont think there is much perf difference switching to unordered_set/map here, so maybe we can update this for all the compilers ?

Reply:
Same comment here: I agree that this would probably be good to do for all compilers, but it might be preferable to keep the changes to a minimum for now to focus on getting the build to work on the different platforms, and we can look at larger changes in the near future.

Reply:
Agreed, that is why I wrap everything into `_MSC_VER `. 
Going forward we can start refactoring the code.